lib/index.js

```diff
@@ -53,37 +56,52 @@ exports.default = function (sails) {
       // to reload sails routes in dev environment builds
       sails.emit('hook:sails-hook-webpack:after-build', rawStats);
 
-      var stats = rawStats.toJson();
       sails.log.debug('sails-hook-webpack: ' + rawStats.toString({ colors: true, chunks: false }));
 
-      if (stats.errors.length > 0) {
-        sails.log.error('sails-hook-webpack: ', stats.errors);
-      }
-
-      if (stats.warnings.length > 0) {
-        sails.log.warn('sails-hook-webpack: ', stats.warnings);
-      }
       return null;
     }
   };
 
   // setup outside like this to allow use of the compiler in http.customMiddleware
-  hook.compiler = (0, _webpack2.default)(config.hook.config, function (err, stats) {
-    if (err) throw err;
-    sails.log.info('sails-hook-webpack: Webpack loaded.');
-    sails.log.silly('sails-hook-webpack: ', stats.toString());
-    if (process.env.NODE_ENV === 'development') {
-      sails.after('lifted', () => {
-        setTimeout(function () {
-          sails.log.info('sails-hook-webpack: Watching for changes...');
-          hook.compiler.watch(config.hook.watchOptions, hook.afterBuild.bind(hook));
-        }, config.watchDelay);
+  if (process.env.NODE_ENV === 'development') {
+      const thread = spawn(function() {});
+      thread.run(function setUpWatch(input, done) {
+
+        var _webpack = require('webpack');
+        var _webpack2 = _interopRequireDefault(_webpack);
+        
+        //change path of webpack config file according to project
+        var config = require('../../../../config/webpack.config.dev.js');
+        
+        function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
+        console.log("Build started", new Date().getTime());
+        
+        var a = {};
+        a.compiler = (0, _webpack2.default)(config.webpack.config, function (err, stats) { 
+          console.log("Building Done", new Date().getTime());
+          done(err, stats.toString());
+          console.log('sails-hook-webpack: Watching for changes...');
+          setTimeout(function() {
+            a.compiler.watch(config.webpack.watchOptions, function() { 
+              console.log("sails-hook-webpack (Watch): Files mapped/changed", new Date().getTime());
+            });
+          }, 1000);
+        });
+      })
+      .send()
+      .on('done',function(err,stats){
+        hook.afterBuild.call(hook, err, stats);
```

Comment:
You dont need after build here. 

Keep the old after build function as it is.

Reply:
Made the required changes.